src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/BlockExpression.cs

```diff
@@ -101,10 +101,10 @@ public BlockExpression Update(IEnumerable<ParameterExpression> variables, IEnume
                 }
             }
 
-            return Block(Type, variables, expressions);
+            return Block(Type, variables, expressions!);
```

Comment:
Even `expression` parameter is non null because of the `if (expressions != null)` check above it became nullable at this point, maybe we should remove this check